pkg/collector/autodiscovery/configresolver.go

```diff
@@ -374,7 +370,7 @@ func parseTemplateVar(v []byte) (name, key []byte) {
 		}
 		return r
 	}, v)
-	split := bytes.Split(stripped, []byte("_"))
+	split := bytes.SplitN(stripped, []byte("_"), 2)
```

Comment:
Do you if there is a difference between 
```go
split := bytes.SplitN(stripped, []byte("_"), 2)
```
and
```go
split := bytes.SplitN(stripped, []byte{'_'}, 2)
```

Reply:
to me, they look equivalent